Lib/fnmatch.py

```diff
@@ -135,18 +136,31 @@ def _translate(pat, star, question_mark):
                         if chunks[k-1][-1] > chunks[k][0]:
                             chunks[k-1] = chunks[k-1][:-1] + chunks[k][1:]
                             del chunks[k]
+
+                    if len(chunks)>1:
+                        char_range=set(range(ord(chunks[0][-1]), ord(chunks[-1][0])))
+
+                        question_mark_char=question_mark.replace('\\', '').replace('[', '').replace(']', '').replace('^', '')
+                        question_mark_char=set(map(ord, question_mark_char))
+
+                        if question_mark_char.intersection(char_range):
+                            inside_range=True
```

Comment:
`question_mark_char` is usually a single character (or can it be something else? I don't think so), it doesn't make sense to compute a set. Second, `char_range` is a continous interval, so you don't need to compute everything to check that it intersects with another set (which in this case, the "other" set is a singleton).

To check if the special question mark is in the range, you can do something like:

```py
if question_mark in '\\[]^':
    inside_range = chunks[0][-1] <= question_mark <= chunks[-1][0]
else:
    inside_range = False
```